rectools/models/nn/sasrec.py

```diff
@@ -177,19 +246,8 @@ def forward(
         """
         seqs *= timeline_mask  # [batch_size, session_max_len, n_factors]
         for i in range(self.n_blocks):
-            q = self.q_layer_norm[i](seqs)
-            mha_output, _ = self.multi_head_attn[i](
-                q, seqs, seqs, attn_mask=attn_mask, key_padding_mask=key_padding_mask, need_weights=False
-            )
-            seqs = q + mha_output
-            ff_input = self.ff_layer_norm[i](seqs)
-            seqs = self.feed_forward[i](ff_input)
-            seqs = self.dropout[i](seqs)
-            seqs += ff_input
-            seqs *= timeline_mask
-
+            seqs = self.transformer_blocks[i](seqs, timeline_mask, attn_mask, key_padding_mask)
```

Comment:
We can stop passing `timeline_mask` to transformer_blocks (here and everywhere). Gonna be more clear.

```
for i in range(self.n_blocks):
  seqs *= timeline_mask
  seqs = self.transformer_blocks[i](seqs, attn_mask, key_padding_mask)
seqs *= timeline_mask
seqs = self.last_layernorm(seqs)
```

Reply:
Fixed